index.md

```diff
@@ -1335,6 +1335,46 @@ One thing for bot authors to note is that the `NOTICE` message may be interprete
 
 ## User-Based Queries
 
+### WHO message
+
+         Command: WHO
+      Parameters: [ <mask> [ "o" ] ]
+
+This command is used to query a list of users who match the provided mask.
+The server will answer this command with zero, one or more [`RPL_WHOREPLY`](#rplwhoreply-352), and end the list with [`RPL_ENDOFWHO`](#rplendofwho-315).
+
+The mask can be one of the following:
+
+* A channel name, in which case the channel members are listed.
+* An exact nickname, in which case a single user is returned.
+* A mask pattern, in which case all visible users whose host, server, real name or nickname matches are listed.
```

Comment:
s/real name/username/ ?

Reply:
This comes from the RFC, which mentions real names instead of usernames. In my testing it seems like real names are matched. Haven't tested usernames.

Reply:
My tests:

* Unreal and Ergo match only channels and usernames
* Solanum and Insp match channels + usernames + realnames, and allow spaces in the mask

Also of note: when the mask contains a space, Unreal/Solanum/Insp all mess up the RPL_ENDOFWHO format by allowing spaces in a non-final argument (ergo returns ERR_UNKNOWNERROR instead). Not sure what they should do instead, though.

Reply:
as usual, explain servers can do whatever they want and clients have to deal with it: "A mask pattern, in which case all visible users whose nickname matches are listed. Servers MAY match other user-specific values, such as usernames, hostnames, or realnames."

Reply:
Definitely willing to reevaluate the Ergo behavior here, although it sounds like the ecosystem is pretty fragmented regardless :-\

The rationale for the current Ergo behavior is, we try to interpret the argument as a NUH mask. (So we can potentially match against nicks, usernames, and hostnames, but never against realnames). If the argument already contains a ! and a @, then the interpretation is straightforward. If it doesn't, we try and expand it to a normal NUH mask with all three components by adding `*`s for the missing components; there may be cases where we guess "wrong" about the expansion. For example, `WHO shivaram` gets expanded to `WHO shivaram!*@*`, i.e. we assume it was intended as a nick rather than a username.

Reply:
> explain servers can do whatever they want and clients have to deal with it

Hm right, done. This isn't as nice for clients, but oh well.

@slingamn I find matching real names to be useful from a client "user search" UI, e.g. I can search "Simon Ser" and end up finding "emersion".